packages/data-provider/src/services/submissionService.ts

```diff
@@ -161,6 +150,112 @@ const service = (dependencies: BaseDependencies) => {
 		});
 	};
 
+	/**
+	 * Validates an Active Submission combined with all Submitted Data.
+	 * Active Submission is updated after validation is compplete.
+	 * Returns the Active Submission updated
+	 * @param {Object} input
+	 * @param {Submission} input.originalSubmission Active Submission
+	 * @param {Record<string, SubmissionData>} input.newSubmissionData New Schema data
+	 * @param {string} input.username User who performs the action
+	 * @returns {Promise<Submission>}
+	 */
+	const performDataValidation = async (input: {
```

Comment:
this is a new function containing logic to merge and validate changes on Active Submission.